video/out/vulkan/context_android.c

```diff
@@ -0,0 +1,128 @@
+/*
+ * This file is part of mpv.
+ *
+ * mpv is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ *
+ * mpv is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with mpv.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include <vulkan/vulkan.h>
+#include <vulkan/vulkan_android.h>
+#include <libavcodec/jni.h>
+#include <android/native_window_jni.h>
+
+#include "common.h"
+#include "options/m_config.h"
+#include "context.h"
+#include "utils.h"
+
+struct android_opts {
+    struct m_geometry surface_size;
+};
+
+extern const struct m_sub_options android_conf; // lol
+
+struct priv {
+    struct mpvk_ctx vk;
+    ANativeWindow *native_window;
+};
+
+static void android_uninit(struct ra_ctx *ctx)
+{
+    struct priv *p = ctx->priv;
+    ra_vk_ctx_uninit(ctx);
+    mpvk_uninit(&p->vk);
+
+    if (p->native_window) {
+        ANativeWindow_release(p->native_window);
+        p->native_window = NULL;
+    }
+}
+
+static bool android_init(struct ra_ctx *ctx)
+{
+    struct priv *p = ctx->priv = talloc_zero(ctx, struct priv);
+    struct mpvk_ctx *vk = &p->vk;
+    int msgl = ctx->opts.probing ? MSGL_V : MSGL_ERR;
+
+    if (!mpvk_init(vk, ctx, VK_KHR_ANDROID_SURFACE_EXTENSION_NAME))
+        goto fail;
+
+    jobject surface = (jobject)(intptr_t)ctx->vo->opts->WinID;
+    JavaVM *vm = (JavaVM *)av_jni_get_java_vm(NULL);
+    JNIEnv *env;
+    int ret = (*vm)->GetEnv(vm, (void**)&env, JNI_VERSION_1_6);
+    if (ret == JNI_EDETACHED) {
+        if ((*vm)->AttachCurrentThread(vm, &env, NULL) != 0) {
+            MP_FATAL(ctx, "Could not attach java VM.\n");
+            goto fail;
+        }
+    }
+    p->native_window = ANativeWindow_fromSurface(env, surface);
+    (*vm)->DetachCurrentThread(vm);
+
+    VkAndroidSurfaceCreateInfoKHR info = {
+         .sType = VK_STRUCTURE_TYPE_ANDROID_SURFACE_CREATE_INFO_KHR,
+         .window = p->native_window
+    };
+
+    VkInstance inst = vk->vkinst->instance;
+    VkResult res = vkCreateAndroidSurfaceKHR(inst, &info, NULL, &vk->surface);
+    if (res != VK_SUCCESS) {
+        MP_MSG(ctx, msgl, "Failed creating Android surface\n");
```

Comment:
Shouldn't this be an `MP_FATAL`?

Reply:
The idea is that it doesn't throw errors in your face when it's just probing, the other contexts do it the identically.